insta/src/macros.rs

```diff
@@ -359,9 +359,44 @@ macro_rules! _assert_snapshot_base {
     // The main macro body — every call to this macro should end up here.
     (transform=$transform:expr, $name:expr, $value:expr, $debug_expr:expr $(,)?) => {
         $crate::_macro_support::assert_snapshot(
-            $name.into(),
-            #[allow(clippy::redundant_closure_call)]
-            &$transform(&$value),
+            (
+                $name,
+                #[allow(clippy::redundant_closure_call)]
+                $transform(&$value).as_str(),
+            ).into(),
+            $crate::_get_workspace_root!().as_path(),
+            $crate::_function_name!(),
+            module_path!(),
+            file!(),
+            line!(),
+            $debug_expr,
+        )
+        .unwrap()
+    };
+}
+
+/// (Experimental)
+/// Asserts a binary snapshot in the form of a [`Vec<u8>`].
+///
+/// The contents get stored in a separate file next to the metadata file. The extension for this
+/// file must be passed as part of the name. For an implicit snapshot name just an extension can be
+/// passed starting with a `.`.
```

Comment:
Great — could we add a couple of examples? This will be the main source of docs for people. And then we can basically copy & paste to the website. (Check out the other macros for examples of examples)

Reply:
Super, thanks!